server/src/test/java/org/elasticsearch/rest/RestRequestTests.java

```diff
@@ -118,6 +118,58 @@ public void testApplyContentParser() throws IOException {
         assertEquals(emptyMap(), source.get());
     }
 
+    public void testParamAsIntWithNoParameters() {
+        RestRequest restRequest = contentRestRequest("", emptyMap());
+        int defaultValue = randomInt();
+        String parameterKey = randomIdentifier();
+
+        int value = restRequest.paramAsInt(parameterKey, defaultValue);
+        assertEquals(defaultValue, value);
+    }
+
+    public void testParamAsIntWithIntegerParameter() {
+        String parameterKey = randomIdentifier();
+        RestRequest restRequest = contentRestRequest("", singletonMap(parameterKey, "123"));
+        int defaultValue = randomInt();
+
+        int value = restRequest.paramAsInt(parameterKey, defaultValue);
+        assertEquals(123, value);
+    }
+
+    public void testParamAsIntWithoutIntegerParameter() {
+        String parameterKey = randomIdentifier();
+        RestRequest restRequest = contentRestRequest("", singletonMap(parameterKey, "123T"));
+        int defaultValue = randomInt();
+
+        assertThrows(IllegalArgumentException.class, () -> { restRequest.paramAsInt(parameterKey, defaultValue); });
```

Comment:
nit: no need for the `{}`, we can use an expression lambda here (also in other cases below)

```suggestion
        assertThrows(IllegalArgumentException.class, () -> restRequest.paramAsInt(parameterKey, defaultValue));
```